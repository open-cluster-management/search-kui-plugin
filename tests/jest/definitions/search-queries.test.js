/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
// Copyright (c) 2021 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'
const searchqueries = require('../../../dist/definitions/search-queries')
const res = require('../../data/pod')
const yaml = require('../../data/deployment')
const savedsearches = require('../../data/savedsearches')

const keys = ['operationName', 'query', 'variables']

const input = {
  filters: [
    {
      property: 'kind',
      value: 'pod'
    }
  ],
  keywords: []
}

// Each query should be defined and return an object with the same keys.
describe('Search Queries', () => {
  it('should DELETE_QUERY', () => {
    expect(searchqueries.DELETE_QUERY(res.pod.name)).toBeDefined()
    expect(Object.keys(searchqueries.DELETE_QUERY(res.pod.name))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.DELETE_QUERY(res.pod.name)).toMatchSnapshot()
  })

  it('should DELETE_RESOURCE', () => {
    expect(searchqueries.DELETE_RESOURCE(res.pod.name, res.pod.namespace, res.pod.kind, res.pod.cluster, res.pod.apiVersion)).toBeDefined()
    expect(Object.keys(searchqueries.DELETE_RESOURCE(res.pod.name, res.pod.namespace, res.pod.kind, res.pod.cluster, res.pod.apiVersion))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.DELETE_RESOURCE(res.pod.name, res.pod.namespace, res.pod.kind, res.pod.cluster, res.pod.apiVersion)).toMatchSnapshot()
  })

  it('should GET_SEARCH_COMPLETE', () => {
    expect(searchqueries.GET_SEARCH_COMPLETE('pod', input)).toBeDefined()
    expect(Object.keys(searchqueries.GET_SEARCH_COMPLETE('pod', input))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.GET_SEARCH_COMPLETE('pod', input)).toMatchSnapshot()
  })

  it('should GET_SEARCH_SCHEMA', () => {
    expect(searchqueries.GET_SEARCH_SCHEMA).toBeDefined()
    expect(Object.keys(searchqueries.GET_SEARCH_SCHEMA)).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.GET_SEARCH_SCHEMA).toMatchSnapshot()
  })

  it('should SAVED_SEARCH_QUERY', () => {
    expect(searchqueries.SAVED_SEARCH_QUERY).toBeDefined()
    expect(Object.keys(searchqueries.SAVED_SEARCH_QUERY)).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.SAVED_SEARCH_QUERY).toMatchSnapshot()
  })

  it('should RESOURCE_LOGS', () => {
    expect(searchqueries.RESOURCE_LOGS(res.pod)).toBeDefined()
    expect(Object.keys(searchqueries.RESOURCE_LOGS(res.pod))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.RESOURCE_LOGS(res.pod)).toMatchSnapshot()
  })

  it('should SEARCH_ACM_QUERY', () => {
    expect(searchqueries.SEARCH_ACM_QUERY(res.pod)).toBeDefined()
    expect(Object.keys(searchqueries.SEARCH_ACM_QUERY(res.pod))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.SEARCH_ACM_QUERY(res.pod)).toMatchSnapshot()
  })

  it('should SEARCH_QUERY', () => {
    expect(searchqueries.SEARCH_QUERY(input.keywords, input.filters)).toBeDefined()
    expect(Object.keys(searchqueries.SEARCH_QUERY(input.keywords, input.filters))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.SEARCH_QUERY(input.keywords, input.filters)).toMatchSnapshot()
  })

  it('should SEARCH_QUERY_COUNT', () => {
    expect(searchqueries.SEARCH_QUERY_COUNT(input)).toBeDefined()
    expect(Object.keys(searchqueries.SEARCH_QUERY_COUNT(input))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.SEARCH_QUERY_COUNT(input)).toMatchSnapshot()
  })

  it('should SEARCH_RELATED_QUERY', () => {
    expect(searchqueries.SEARCH_RELATED_QUERY(input.keywords, input.filters)).toBeDefined()
    expect(Object.keys(searchqueries.SEARCH_RELATED_QUERY(input.keywords, input.filters))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.SEARCH_RELATED_QUERY(input.keywords, input.filters)).toMatchSnapshot()
  })

  it('should UPDATE_RESOURCE', () => {
    expect(searchqueries.UPDATE_RESOURCE(yaml.deployment)).toBeDefined()
    expect(Object.keys(searchqueries.UPDATE_RESOURCE(yaml.deployment))).toEqual(expect.arrayContaining(keys))
    expect(searchqueries.UPDATE_RESOURCE(yaml.deployment)).toMatchSnapshot()
  })

  savedsearches.data.forEach(search => {
    it(`should SAVE_SEARCH query - ${search.name}`, () => {
      expect(searchqueries.SAVE_SEARCH(search)).toBeDefined()
      expect(Object.keys(searchqueries.SAVE_SEARCH(search))).toEqual(expect.arrayContaining(keys))
      expect(searchqueries.SAVE_SEARCH(search)).toMatchSnapshot()
    })
  });
})
