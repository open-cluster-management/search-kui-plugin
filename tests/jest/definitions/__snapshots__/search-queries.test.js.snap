// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Queries should DELETE_QUERY 1`] = `
Object {
  "operationName": "deleteSearch",
  "query": "mutation deleteSearch($resource: JSON!) {
 deleteSearch(resource: $resource)
}
",
  "variables": Object {
    "resource": Object {
      "name": "nginx",
    },
  },
}
`;

exports[`Search Queries should DELETE_RESOURCE 1`] = `
Object {
  "operationName": "deleteResource",
  "query": "mutation deleteResource($apiVersion: String, $name: String, $namespace: String, $cluster: String, $kind: String, $childResources: JSON) {
  deleteResource(apiVersion: $apiVersion, name: $name, namespace: $namespace, cluster: $cluster, kind: $kind, childResources: $childResources)
}
",
  "variables": Object {
    "cluster": "local-cluster",
    "kind": "Pod",
    "name": "nginx",
    "namespace": "kube-system",
    "apiVersion": "v1",
  },
}
`;

exports[`Search Queries should GET_SEARCH_COMPLETE 1`] = `
Object {
  "operationName": "searchComplete",
  "query": "query searchComplete($property: String!, $query: SearchInput) {
  searchComplete(property: $property, query: $query)
 }",
  "variables": Object {
    "property": "pod",
    "query": Object {
      "filters": Array [
        Object {
          "property": "kind",
          "value": "pod",
        },
      ],
      "keywords": Array [],
    },
  },
}
`;

exports[`Search Queries should GET_SEARCH_SCHEMA 1`] = `
Object {
  "operationName": "searchSchema",
  "query": "query searchSchema {
  searchSchema
  }",
  "variables": Object {},
}
`;

exports[`Search Queries should RESOURCE_LOGS 1`] = `
Object {
  "operationName": "getLogs",
  "query": "query getLogs($containerName: String!, $podName: String!, $podNamespace: String!, $clusterName: String!) {
  logs(containerName: $containerName, podName: $podName, podNamespace: $podNamespace, clusterName: $clusterName)
}
",
  "variables": Object {
    "clusterName": "local-cluster",
    "containerName": Array [
      "fake-container",
    ],
    "podName": "nginx",
    "podNamespace": "kube-system",
  },
}
`;

exports[`Search Queries should SAVE_SEARCH query - All deployments 1`] = `
Object {
  "operationName": "saveSearch",
  "query": "mutation saveSearch($resource: JSON!) {
  saveSearch(resource: $resource)
}
",
  "variables": Object {
    "resource": Object {
      "description": "All deployments",
      "id": "1573570796793",
      "name": "All deployments",
      "searchText": "kind:deployments",
    },
  },
}
`;

exports[`Search Queries should SAVE_SEARCH query - All pods 1`] = `
Object {
  "operationName": "saveSearch",
  "query": "mutation saveSearch($resource: JSON!) {
  saveSearch(resource: $resource)
}
",
  "variables": Object {
    "resource": Object {
      "description": "All pods",
      "id": "1573570796760",
      "name": "All pods",
      "searchText": "kind:pod",
    },
  },
}
`;

exports[`Search Queries should SAVED_SEARCH_QUERY 1`] = `
Object {
  "operationName": "savedSearches",
  "query": "query savedSearches {
  items: savedSearches {
    id
    name
    description
    searchText
    __typename
  }
}
",
  "variables": Object {},
}
`;

exports[`Search Queries should SEARCH_ACM_QUERY 1`] = `
Object {
  "operationName": "getResource",
  "query": "query getResource($kind: String, $name: String, $namespace: String, $cluster: String, $apiVersion: String) {
  getResource(kind: $kind, name: $name, namespace: $namespace, cluster: $cluster, apiVersion: $apiVersion)
}
",
  "variables": Object {
    "cluster": "local-cluster",
    "kind": "Pod",
    "name": "nginx",
    "namespace": "kube-system",
    "apiVersion": "v1",
  },
}
`;

exports[`Search Queries should SEARCH_QUERY 1`] = `
Object {
  "operationName": "searchResult",
  "query": "query searchResult($input: [SearchInput]) {
  searchResult: search(input: $input) {
    items
    __typename
  }
}
",
  "variables": Object {
    "input": Array [
      Object {
        "filters": Array [
          Object {
            "property": "kind",
            "value": "pod",
          },
        ],
        "keywords": Array [],
      },
    ],
  },
}
`;

exports[`Search Queries should SEARCH_QUERY_COUNT 1`] = `
Object {
  "operationName": "searchResult",
  "query": "query searchResult($input: [SearchInput]) {
  searchResult: search(input: $input) {
    count
    __typename
  }
}
",
  "variables": Object {
    "input": Object {
      "filters": Array [
        Object {
          "property": "kind",
          "value": "pod",
        },
      ],
      "keywords": Array [],
    },
  },
}
`;

exports[`Search Queries should SEARCH_RELATED_QUERY 1`] = `
Object {
  "operationName": "searchResult",
  "query": "query searchResult($input: [SearchInput]) {
  searchResult: search(input: $input) {
    items
    related {
      kind
      items
      __typename
    }
    __typename
  }
}
",
  "variables": Object {
    "input": Array [
      Object {
        "filters": Array [
          Object {
            "property": "kind",
            "value": "pod",
          },
        ],
        "keywords": Array [],
      },
    ],
  },
}
`;

exports[`Search Queries should UPDATE_RESOURCE 1`] = `
Object {
  "operationName": "updateResource",
  "query": "query updateResource($apiVersion: String, $namespace: String, $kind: String, $name: String, $body: JSON, $cluster: String) {
  updateResource(apiVersion: $apiVersion, namespace: $namespace, kind: $kind, name: $name, body: $body, cluster: $cluster)
}
",
  "variables": Object {
    "body": Object {
      "apiVersion": "apps/v1",
      "cluster": "local-cluster",
      "kind": "Deployment",
      "metadata": Object {
        "labels": Object {
          "app": "drone-app",
        },
        "name": "myapp",
        "namespace": "kube-system",
      },
      "name": "myapp",
      "namespace": "kube-system",
      "spec": Object {
        "replicas": 1,
        "selector": Object {
          "matchLabels": Object {
            "app": "drone-app",
          },
        },
        "template": Object {
          "metadata": Object {
            "labels": Object {
              "app": "drone-app",
            },
          },
          "spec": Object {
            "affinity": Object {
              "podAntiAffinity": Object {
                "requiredDuringSchedulingIgnoredDuringExecution": Array [
                  Object {
                    "labelSelector": Object {
                      "matchExpressions": Array [
                        Object {
                          "key": "app",
                          "operator": "In",
                          "values": Array [
                            "drone-app",
                          ],
                        },
                      ],
                    },
                    "topologyKey": "kubernetes.io/hostname",
                  },
                ],
              },
            },
            "containers": Array [
              Object {
                "image": "rvennam/drone-app:latest",
                "name": "drone-app",
                "ports": Array [
                  Object {
                    "containerPort": 3000,
                  },
                ],
                "resources": Object {
                  "limits": Object {
                    "cpu": "100m",
                    "memory": "128Mi",
                  },
                  "requests": Object {
                    "cpu": "50m",
                    "memory": "64Mi",
                  },
                },
              },
            ],
            "terminationGracePeriodSeconds": 0,
          },
        },
      },
    },
    "cluster": "local-cluster",
    "kind": "Deployment",
    "name": "myapp",
    "namespace": "kube-system",
  },
}
`;
